File: Two Sums/Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
    const numLength = nums.length
    let indexValueDict = {}
    for (let i = 0; i < numLength; i++){
        let complement = target - nums[i]
        if(complement in indexValueDict)
            return [i, indexValueDict[complement]]
        indexValueDict[nums[i]] = i
    }      
};

console.log(twoSum(nums = [3,5,9,7], target = 16))

File: Two Sums/Two Sum.py
def twoSum(nums: list[int], target: int) -> list[int]:
    indexValueDict = {}
    for i in range(len(nums)):
        if((target - nums[i]) in indexValueDict):
            return [i, indexValueDict[target - nums[i]]]
        indexValueDict[nums[i]] = i
        
print(twoSum(nums=[2,5,6,7], target=13))
        
        
        
        
 